pvlib/modelchain.py

```diff
@@ -931,23 +996,125 @@ def prepare_inputs(self, weather, times=None):
             airmass=self.airmass['airmass_relative'],
             model=self.transposition_model)
 
-        if self.weather.get('wind_speed') is None:
-            self.weather['wind_speed'] = 0
-        if self.weather.get('temp_air') is None:
-            self.weather['temp_air'] = 20
+        return self
+
+    def prepare_inputs_from_poa(self, data):
+        """
+        Prepare the solar position, irradiance, and irradiance inputs to
+        the model, starting with plane-of-array irradiance.
+
+        Parameters
+        ----------
+        data : DataFrame
+            Contains plane-of-array irradiance data. Required column names
+            include ``'poa_global'``, ``'poa_direct'`` and ``'poa_diffuse'``.
+            Columns with weather-related data are ssigned to the
+            ``weather`` attribute.  If columns for ``'temp_air'`` and
+            ``'wind_speed'`` are not provided, air temperature of 20 C and wind
+            speed of 0 m/s are assumed.
+
+        Notes
+        -----
+        Assigns attributes: ``weather``, ``total_irrad``, ``solar_position``,
+        ``airmass``, ``aoi``, ``temperature``.
+
+        See also
+        --------
+        pvlib.modelchain.ModelChain.prepare_inputs
+        """
+
+        self._assign_weather(data)
+
+        self._verify_df(data, req=['poa_global', 'poa_direct', 'poa_diffuse'])
+        self._assign_total_irrad(data)
+
+        self._prep_inputs_solar_pos()
+        self._prep_inputs_airmass()
+
+        if isinstance(self.system, SingleAxisTracker):
+            self._prep_inputs_tracking()
+        else:
+            self._prep_inputs_fixed()
+
+        return self
+
+    def prepare_temperature(self, data=None):
+        """
+        Sets cell_temperature using inputs in data and the specified
+        temperature model.
+
+        If 'data' contains 'cell_temperature', these values are assigned to
+        attribute ``cell_temperature``. If 'data' contains 'module_temperature`
+        and `temperature_model' is 'sapm', cell temperature is calculated using
+        :py:func:`pvlib.temperature.sapm_celL_from_module`. Otherwise, cell
+        temperature is calculated by 'temperature_model'.
+
+        Parameters
+        ----------
+        data : DataFrame, default None
+            May contain columns ``'cell_temperature'`` or
+            ``'module_temperaure'``.
+
+        Returns
+        -------
+        self
+
+        Assigns attribute ``cell_temperature``.
+
+        """
+        try:
+            self.cell_temperature = data['cell_temperature']
+            if self.temperature_model is not None:
+                warnings.warn('using ''cell_temperature'' provided in input',
+                              ' rather than specified temperature model: {0}'
+                              .format(self.temperature_model.__name__))
+            return self
+        except:
+            pass
```

Comment:
We don't currently allow `self.temperature_model` to be `None`. In any case, I don't see why this warning would be helpful.